### PR TITLE
feat(desktop): wire @muyajs/core parity APIs (menu state, copyAsRich, heading-link, source cursor, saved indicator)

### PR DESCRIPTION
**Pull request by Jocs**
## Summary

Final wave of the muyajs → @muyajs/core parity follow-ups to #4406. The seven Wave-1 engine PRs (#4408–#4414) landed the engine halves; this PR wires the **desktop** consumers and flips the remaining desktop scoreboard tests. **14 of the 15 confirmed gaps are now closed; PG14 is accept-deferred.**

## Per-gap outcome

| Gap | What this PR wires | Status |
|-----|--------------------|--------|
| **PG1** | `adaptSelectionChange` consumes the new `affiliation` chain + `anchorBlockInfo`/`focusBlockInfo` (engine #4410); fixes the consumer's loose-list read. Native Paragraph/Format menu check-marks, loose/task-list toggles, table/code detection, and Format-disable-in-code work again. | ✅ fixed |
| **PG2** | New engine helper `Muya#setCursorByOffset` + desktop `handleFileChange` maps the saved CodeMirror `{line, ch}` `muyaIndexCursor` onto a block-key cursor, restoring the WYSIWYG caret after a source-mode edit. | ✅ fixed (engine helper added) |
| **PG8** | `util/pdf.ts::getHtmlToc` swaps the legacy muyajs `Slugger` for `@muyajs/core`'s `generateGithubSlug`, replicating the engine's whole-document `-N` dedup (#4412) so exported-TOC anchors match the injected heading ids. | ✅ fixed |
| **PG9** | `COPY_PASTE_METHOD_MAP.copyAsRich` → the real `Muya#copyAsRich` (engine #4411). "Copy as Rich Text" now yields formatted rich text, not HTML source. | ✅ fixed |
| **PG10** | `preview-image` subscription already present; engine now emits it (#4414). **Verified, no change.** | ✅ fixed |
| **PG11** | Re-added `editor.on('heading-copy-link', ({ key }) => editorStore.copyGithubSlug(key))` (engine emits it, #4414). | ✅ fixed |
| **PG15** | `makeSyntheticHistory` id is now the engine undo-stack **depth** (a stable position marker, not an ever-incrementing counter), and a freshly-loaded tab seeds `lastSavedHistoryId` to the baseline depth (0). Undo-to-disk restores the saved/clean indicator. | ✅ fixed |
| **PG14** | First undo after source mode reverting the bulk edit in one step. **Accept-deferred** — see below. | ⏸️ deferred |

PG3/PG12 (#4409), PG4 (#4413), PG5/PG6 (#4411), PG7 (#4412), PG13 (#4408) are engine-only and needed no desktop change.

## Engine helper added

`packages/muya/src/selection/offsetCursor.ts` + `Muya#setCursorByOffset(indexCursor)` (separate commit `feat(muya): add setCursorByOffset…`). It reproduces the legacy muyajs index→block-key conversion: inject sentinel strings into the current markdown at the line/ch offsets, rebuild the tree, find the content blocks they landed in, then rebuild the clean document and set the cursor by the resolved paths + offsets (both `setContent` calls run synchronously, so no intermediate paint). Covered by `setCursorByOffset.spec.ts` (5 `PG2:` cases).

## PG14 — accept-defer rationale

On source-mode exit, `handleFileChange` rebuilds the doc via `setContent` (which `history.clear()`s) then restores the pre-source op stack, so the bulk change is not a single undo boundary. Recording it as one boundary would require computing a general whole-document `ot-json1` diff and feeding it through `Editor.updateContents`' pick/drop walker, which only handles a fixed set of op shapes (block insert at index, text edit, `checked`/`meta`); an arbitrary diff (removes/moves/nested replaces) risks corrupting the document. The risk outweighs the benefit (narrow first-undo-granularity edge case; undo still works, prior stack intact), so PG14 is left `test.fail()` with explanatory notes in `handleFileChange` and the spec. A clean revival needs a dedicated engine "record a state replacement as one op" API.

## Tests

- Flipped `test.fail()` → real on PG1 (`parity-pg1-menu-state.spec.ts`), PG2 + PG15 (`parity-source-undo-saved.spec.ts`); all genuinely pass.
- PG14 stays `test.fail()`.
- Added `packages/muya/src/__tests__/setCursorByOffset.spec.ts`.
- Reconciled `packages/desktop/test/PARITY_SCOREBOARD.md` to the true state (gaps remaining 11 → 1).

## Verification

`pnpm typecheck` (only the pre-existing `electron.vite.config.ts` vite/rolldown typing error remains), `pnpm lint` (0 errors), `pnpm test` (560 pass), `pnpm build:unpack` (clean), the parity e2e suite (4 pass incl. PG14 xfail) + core e2e (launch/editor-input/tabs/selection-change/inline-format/paragraph-blocks, 36 pass), and the full muya engine gauntlet (lint/types/circular/553 unit/1347 spec — all green).

Parity follow-up to #4406.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/desktop/src/renderer/src/components/editorWithTabs/editor.vue [hunk 1]
```diff
@@ -297,38 +307,77 @@ interface SelectionFormatLike {
...
+    return functionType ? { ...entry, functionType } : entry
+  })
   return {
     start: {
       key: anchorPath.join('/'),
       offset: (changes.anchor?.offset ?? 0) as number,
-      block: anchorBlock,
-      type: changes.type as string | undefined
+      block: { functionType: anchorInfo?.functionType },
+      type: anchorInfo?.type
```
[the remark this answers is not in the document]
Comment (Jocs) [reply packages/desktop/src/renderer/src/components/editorWithTabs/editor.vue:378]: Good catch — fixed in 5fe91bd. `adaptSelectionChange` now carries `block.text` from the live `anchorBlock`/`focusBlock` again, so `SELECTION_CHANGE`'s `start.block.text.substring(...)` (search prefill) keeps working. Verified by find-replace + selection-change e2e.

File: packages/desktop/src/renderer/src/components/editorWithTabs/editor.vue [hunk 2]
```diff
@@ -1255,12 +1305,29 @@ interface FileChangePayload {
...
+): cursor is { anchor: { line: number; ch: number }; focus: { line: number; ch: number } } => {
+  const c = cursor as { anchor?: { line?: unknown }; focus?: { line?: unknown } } | null
+  return (
+    !!c &&
+    !!c.anchor &&
+    !!c.focus &&
+    typeof c.anchor.line === 'number' &&
+    typeof c.focus.line === 'number'
+  )
+}
```
[the remark this answers is not in the document]
Comment (Jocs) [reply packages/desktop/src/renderer/src/components/editorWithTabs/editor.vue:1328]: Fixed in 5fe91bd. Factored out `isIndexPosition` and `isIndexCursor` now requires both `line` AND `ch` to be numbers, so a missing/invalid `ch` no longer falls through to the column-0 clamp.

File: packages/muya/src/muya.ts [hunk 3]
```diff
@@ -717,6 +719,40 @@ export class Muya {
...
+     * index cursor (PG2 parity). The block tree has no source-line mapping, so
+     * the offsets are resolved the way legacy muyajs did: inject sentinel
+     * strings into the current markdown at the line/ch positions, rebuild the
+     * tree (sentinels embed as literal text), find which content blocks they
+     * landed in, then rebuild the clean document and set the cursor by the
+     * resolved block paths + offsets. The sentinel-bearing tree is transient —
+     * both `setContent` calls run synchronously within this task, so no
+     * intermediate paint happens. No-op (returns `false`) when the cursor is
+     * stale / unresolvable, letting the caller fall back to its default.
+     */
```
[the remark this answers is not in the document]
Comment (Jocs) [reply packages/muya/src/muya.ts:738]: Fixed in 5fe91bd. `setCursorByOffset` now snapshots `getHistory()` and restores it after the internal `setContent` rebuild, so it is a caret-only operation that preserves the undo stack. Documented in the JSDoc and covered by a new history-preservation test.
